File: practce_new/coding-test-sample-server/sample.py
import sys
import requests

def main():
    if len(sys.argv) < 3:
        print("Usage: python script.py <accessToken> <keyword>")
        sys.exit(1)

    access_token = sys.argv[1]
    keyword = sys.argv[2]
    url = 'https://challenge-server.tracks.run/hotel-reservation/hotels'  # エンドポイントURL

    # URLにクエリパラメータを追加
    params = {'keyword': keyword}
    headers = {'X-ACCESS-TOKEN': access_token}

    try:
        # APIリクエストを送信
        response = requests.get(url, params=params, headers=headers)
        response.raise_for_status()  # エラーが発生した場合に例外を発生させる

        # レスポンスをJSONとしてデコード
        data = response.json()

        # ホテル名を表示
        for hotel in data:
            print(hotel['name'])

    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: practice_new/coding-test-sample-server/sample.php
<?php

function main ($argc, $argv) {
  $accessToken = $argv[0];
  $keyword = $argv[1];
  $url = 'https://challenge-server.tracks.run/hotel-reservation/hotels'; // エンドポイントURL

  // URLにクエリパラメータを追加
  $queryParams = http_build_query(['keyword' => $keyword]);
  $requestUrl = "{$url}?{$queryParams}";

  // cURLの初期化
  $ch = curl_init();

  // cURLオプションの設定
  curl_setopt($ch, CURLOPT_URL, $requestUrl);
  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
  curl_setopt($ch, CURLOPT_HTTPHEADER, [
      "X-ACCESS-TOKEN: $accessToken"
  ]);

  // APIのレスポンスを取得
  $response = curl_exec($ch);
  curl_close($ch);

  // レスポンスをJSONとしてデコード
  $data = json_decode($response, true);

  foreach($data as $hotel) {
    echo $hotel['name'] . "\n";
  }
}

main($argc - 1, array_slice($argv, 1));
